pylint/checkers/base.py

```diff
@@ -2511,14 +2511,18 @@ def _check_callable_comparison(self, node):
         left_operand, right_operand = node.left, node.ops[0][1]
         # this message should be emitted only when there is comparison of bare callable
         # with non bare callable.
-        if (
-            sum(
-                1
-                for operand in (left_operand, right_operand)
-                if isinstance(utils.safe_infer(operand), bare_callables)
-            )
-            == 1
-        ):
+        number_of_bare_callables = 0
+        for operand in left_operand, right_operand:
```

Comment:
I did not know you do not need to cast to tuple here. Learn something new every day :)

Reply:
Just like `for k, v in d.items()`! Or its evil twin `for k, in d.items()` that I worked on in #5505

Reply:
😄